people_tool/tests/jasmine/SearchControllerTests.js

```diff
@@ -51,15 +47,135 @@ describe('Unit testing people_tool SearchController', function() {
         });
     });
 
-    describe('searchPeople', function() {
-        beforeEach(setupController);
-        it('sends expected GET params to backend');
-        it('updates scope with no error message when backend returns no error');
-        it('updates scope error message when backend returns error');
+    describe('searchPeople', function () {
+        beforeEach(function() {
+            scope.dtInstance = jasmine.createSpyObj('dtInstance', ['reloadData']);
+        });
+
+        it('does not hit backend if search string is too long', function() {
+            // at controller initialization, default search term has a maximum
+            // length of 32 characters
+            scope.queryString = '123456789012345678901234567890123';
+            scope.searchPeople();
+            expect($timeout.flush).toThrowError('No deferred tasks to be flushed');
+            expect(scope.dtInstance.reloadData).not.toHaveBeenCalled();
+            expect(scope.messages.length).toBe(1);
+            expect(scope.messages[0].type).toEqual('warning');
```

Comment:
Ok - I can look into striking a balance between generic and specific by checking each of the `searchType`s.